plugins/cue_points/base/lib/kCuePointManager.php

```diff
@@ -249,6 +260,32 @@ public function objectReplaced(BaseObject $object, BaseObject $replacingObject,
 		return true;
 	}
 
+	/* (non-PHPdoc)
+	 * @see kObjectCopiedEventConsumer::objectCopied()
+	 */
+	public function objectCopied(BaseObject $fromObject, BaseObject $toObject)
+	{
+		if($fromObject instanceof entry)
+		{
+			$c = new KalturaCriteria();
+			$c->add( CuePointPeer::ENTRY_ID, $fromObject->getId() );
+			$c->addAscendingOrderByColumn(CuePointPeer::CREATED_AT);
+			$cpToCopy = CuePointPeer::doSelect($c);
+			if ( count( $cpToCopy ) <= self::MAX_CUE_POINTS_TO_COPY )
```

Comment:
You should count the cue-points that copied and not the cue-points that returned from the select.
For example, you may have 10,000 answer cue-point that will be skipped in shouldCloneByProperty, and it will cause that no other cue-point types will be copied.